docs/arch.md

```diff
@@ -285,30 +285,28 @@ Such a client is used in [GatherMachineSet](pkg/gather/clusterconfig/clusterconf
-1. the 'adapter-part' that is called by the `Gatherer.Gather`, named `Gather<Something>`, it handles the creation of the clients and handling the communication with the `Gatherer`.
-2. the 'core-part' that holds the actual logic of what to gather, named `gather<Something>`, the clients required for this are passed in as arguments by the 'adapter-part'.
-
-Gather-functions are IO bound and they don't use much of the CPU, so giving each of them a goroutine doesn't stress the CPU but gives us an 'async' way of making REST calls, which improves the performance greatly.
-
-After starting the goroutines, the Gatherer will start monitoring the channels, when it receives a result it will:
-- Store the received `record`s using the provided `record.Interface`'s `Record` method.
-- Store some metadata about the gather-function.
-- Collect the errors accordingly. Errors are accumulated over all the gather-functions and returned as 1 summed up error.
-
-Each result is being stored into record.Item as Marshalable item. It is using either golang Json marshaller, or K8s Api serializers. Those has to be explicitly registered in init func. The record is created to archive under its Name specifying full relative path including folders. The extension for particular record file is defined by GetExtension() func, but most of them are today of "json", except metrics or id.
-
-The `gatherFunctions` map is where we reference all the gather-functions we have within the `clusterconfig` package.
-Each has an id (the key in the map) these can be used to only execute a selection of the gather-functions. (according to the default config all gather-function will be executed)
-Furthermore each gather-function is categorized into either:
-- `important` meaning if that gather-function has an error we will notify `periodic.go` about it, which will handle it accordingly.
-- `failable` meaning if that gather-function has an error we will just log it and add it to our metadata.
-This is necessary as we are expanding into gathering data about resources that are not guaranteed to be present on the cluster. By default if a resource is not present we shouldn't see an error, but it's better to be safe.
+Insights operator defines three types of gatherers (see below). Each of them must implement the [Interface](https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/interface.go#L11) and there are three types of gatherers (see below). These are initialized by calling [gather.CreateAllGatherers](https://github.com/openshift/insights-operator/blob/master/pkg/controller/operator.go#L124) in `operator.go`. The actual gathering is triggered in [periodic.go](https://github.com/openshift/insights-operator/blob/master/pkg/controller/periodic/periodic.go#L93), but not every gatherer is triggered every time ( for example, see the [CustomPeriodGatherer type](https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/interface.go#L21)).
```

Comment:
@tremes I just noticed your edit duplicates "three types of gatherers (see below)" -was this intentional?

Reply:
Ohhh nice catch! Thanks.